bcdaworker/main.go

```diff
@@ -293,5 +295,61 @@ func main() {
 	fmt.Println("Starting bcdaworker...")
 	workerPool := setupQueue()
 	defer workerPool.Close()
+	expiredScheduler()
 	waitForSig()
 }
+
+func updateStatus(hrThreshold int) {
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	var jobs []models.Job
+	err := db.Find(&jobs, "status = ?", "Completed").Error
+	if err != nil {
+		log.Error(err)
+	}
+
+	expDir := os.Getenv("FHIR_EXPIRED_DIR")
+	if _, err = os.Stat(expDir); os.IsNotExist(err) {
+		err = os.MkdirAll(expDir, os.ModePerm)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
+	for _, j := range jobs {
+		t := j.CreatedAt
+		elapsed := time.Since(t).Hours()
+		if int(elapsed) >= hrThreshold {
+			j.Status = "Expired"
+			err = db.Save(j).Error
+			if err != nil {
+				log.Error(err)
+			}
+
+			id := strconv.Itoa(int(j.ID))
+			jobDir := fmt.Sprintf("%s/%s", os.Getenv("FHIR_PAYLOAD_DIR"), id)
+			expDir = fmt.Sprintf("%s/%s", os.Getenv("FHIR_EXPIRED_DIR"), id)
+
+			err = os.Rename(jobDir, expDir)
+			if err != nil {
+				log.Error(err)
+			}
+		}
+	}
+}
+
+func expiredScheduler() {
```

Comment:
We could very easily use `crontab` handle scheduling and execution of this routine. That would mean we could drop the `github.com/robfig/cron` dependency. It would require refactoring this to be a CLI command instead.

Reply:
Okay cool, I'm unfamiliar but I'll definitely look into cron tab

Reply:
Its something we would configure on the instance and not within the app.  When it triggers on the system, it would invoke the app to do the file moving.

Reply:
https://docs.ansible.com/ansible/2.5/modules/cron_module.html